src/lib/Libutil/misc_utils.c

```diff
@@ -715,6 +715,66 @@ replace(char *str, char	*sub, char *repl, char	*retstr)
 	strncpy(retstr, rstr, i + 1);
 }
 
+
+/**
+ * @brief
+ *	Add esc as escape character if delim is appearing on str
```

Comment:
shall we rephrase this to - Escape every occurence of 'delim' in 'str' with 'esc'.
Same for the header file where we declare this function.